Lib/test/test_ast/test_ast.py

```diff
@@ -3035,3 +3035,135 @@ def test_cli_file_input(self):
         self.assertEqual(expected.splitlines(),
                          res.out.decode("utf8").splitlines())
         self.assertEqual(res.rc, 0)
+
+
+class ASTOptimiziationTests(unittest.TestCase):
+    binop = {
+        "+": ast.Add(),
+        "-": ast.Sub(),
+        "*": ast.Mult(),
+        "/": ast.Div(),
+        "%": ast.Mod(),
+        "<<": ast.LShift(),
+        ">>": ast.RShift(),
+        "|": ast.BitOr(),
+        "^": ast.BitXor(),
+        "&": ast.BitAnd(),
+        "//": ast.FloorDiv(),
+        "**": ast.Pow(),
+    }
+
+    unaryop = {
+        "~": ast.Invert(),
+        "+": ast.UAdd(),
+        "-": ast.USub(),
+    }
+
+    def find_node(self, code, optimize, target):
+        tree = ast.parse(code, optimize=optimize)
+        for node in ast.walk(tree):
+            if ast.compare(node, target):
+                return True
+        return False
+
+    def create_binop(self, operand):
+        return ast.BinOp(left=ast.Constant(1), op=self.binop[operand], right=ast.Constant(1))
+
+    def create_unaryop(self, operand):
+        return ast.UnaryOp(op=self.unaryop[operand], operand=ast.Constant(1))
+
+    def assert_ast(self, code, non_optimized_target, optimized_target):
+        # Searching for a non-optimized node in an unoptimized AST
+        self.assertTrue(self.find_node(code, optimize=-1, target=non_optimized_target))
+
+        # Searching for a non-optimized node in an optimized AST
+        self.assertFalse(self.find_node(code, optimize=1, target=non_optimized_target))
+
+        # Searching for a constant node in an optimized AST
+        self.assertTrue(self.find_node(code, optimize=1, target=optimized_target))
+
+    def test_folding_binop(self):
+        code = "1 %s 1"
+        operators = self.binop.keys()
+
```

Comment:
You may move create_binop() here, as a nested function, since it's only used here.

Reply:
Thanks, I've also done the same thing for ``create_unaryop()``